gap/algebra.gi

```diff
@@ -148,15 +148,8 @@ BindGlobal("TOTALDEGREE@", function(x)
 end);
 
 BindGlobal("STRINGSTOLMACHINE@", function(r,args,creator)
-    local temp, i, j, gens, transitions, output, data, Error;
+    local temp, i, j, gens, transitions, output, data;
 
-    Error := function(arg)
-        if IsBound(data) then
-            MakeReadWriteGlobal(data.holdername); Unbind(data.holdername);
-        fi;
-        CallFuncList(VALUE_GLOBAL("Error"),arg);
-    end;
```

Comment:
Another issue with the old code: this trick here to fake installing an exception handler of course only works with `Error` calls made directly in this function. But if any code called by it runs into an error (such as the code being evaluated from a string), then this is not triggered.